mlflow/tensorflow/__init__.py

```diff
@@ -782,6 +797,39 @@ def train(original, self, *args, **kwargs):
 
         result = original(self, *args, **kwargs)
 
+        def _extract_data_from_input_fn():
```

Comment:
Can we move this function into `_autolog.py` and have it accept training data input as an argument? Then, can we make the methods it relies on private by prefixing with `_`?  E.g. `extract_sample_dict_or_tensor` -> `_extract_sample_dict_or_tensor`.